jupyterhub/proxy.py

```diff
@@ -26,6 +28,54 @@
 from .utils import url_path_join
 
 
+class RouteSpec(namedtuple('RouteSpec', ['path', 'host'])):
+    """Object wrapping a proxy route specification
+
+    Attributes:
+
+        path (str): The URL path prefix for this route. Required.
+        host (str): The hostname used for routing when host-based routing is enabled.
+
+    """
+    def __new__(cls, path, *, host=''):
```

Comment:
How could `''` have a meaning as a host when it's not a valid hostname? If host is not specified, the Proxy implementation (the Python API here) should not send host info to the upstream implementation at all. It's a matter for JupyterHub-specific Python code, not the upstream implementation.

I have a slight preference to '' over None so that the type is stable, but None is okay, too. I don't think we should use '*' because `bool(host)` should be False when host is not specified.